library/src/main/scala/scala/scalajs/runtime/RuntimeString.scala

```diff
@@ -166,6 +166,32 @@ private[runtime] object RuntimeString {
     Pattern.matches(regex, thiz)
   }
 
+  // Both regionMatches ported from https://github.com/gwtproject/gwt/blob/master/user/super/com/google/gwt/emul/java/lang/String.java
+  def regionMatches(thiz: String, ignoreCase: Boolean,
+                    toffset: Int, other: String,
+                    ooffset: Int, len: Int): Boolean = {
+    if (other == null) {
```

Comment:
Sure, np. But I notice that not all methods have this check, is that deliberate or not?